src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,206 @@
+            if (result.next()) {
+                car.setId(result.getObject(1, Long.class));
+            }
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can`t insert to DB, car -" + car, throwables);
+        }
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String selectQuery = "SELECT * FROM cars c JOIN manufacturers m"
+                + " ON c.manufacturer_id = m.id WHERE c.id = ? AND c.is_deleted = FALSE;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement getCarPreparedStatement = connection
+                     .prepareStatement(selectQuery)) {
+            getCarPreparedStatement.setLong(1, id);
+            ResultSet rs = getCarPreparedStatement.executeQuery();
+            if (rs.next()) {
+                car = parseCarFromResultSet(rs);
+            }
+        } catch (SQLException throwables) {
+            throwables.printStackTrace();
+        }
+        if (car != null) {
+            car.setDrivers(getDriversForCar(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        List<Car> cars = new ArrayList<>();
+        String getAllQuery = "SELECT * FROM cars c JOIN manufacturers m"
+                + " ON c.manufacturer_id = m.id WHERE c.is_deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement getAllCarsPreparedStatement = connection
+                     .prepareStatement(getAllQuery)) {
+            ResultSet rs = getAllCarsPreparedStatement.executeQuery();
+            while (rs.next()) {
+                cars.add(parseCarFromResultSet(rs));
+            }
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can`t get all cars from DB" + throwables);
+        }
+        cars.forEach(car -> car.setDrivers(getDriversForCar(car.getId())));
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateQuery = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement updateCarsPreparedStatement = connection
+                     .prepareStatement(updateQuery)) {
+            updateCarsPreparedStatement.setString(1, car.getModel());
+            updateCarsPreparedStatement.setLong(2, car.getManufacturer().getId());
+            updateCarsPreparedStatement.setLong(3, car.getId());
+            updateCarsPreparedStatement.executeUpdate();
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can`t update car:" + car, throwables);
+        }
+        insertDrivers(car);
+        removeDriver(car.getId());
```

Comment:
maybe should first delete old drivers, then add new drivers.
```suggestion
        removeDriver(car.getId());
        insertDrivers(car);
```